content/master/concepts/patch-and-transform.md

```diff
@@ -0,0 +1,1692 @@
+* `ToJson` - Convert the input string to valid JSON.
+* `ToSha1` - Create a SHA-1 hash of the input string.
+* `ToSha256` - Create a SHA-256 hash of the input string.
+* `ToSha512` - Create a SHA-512 hash of the input string. 
+
+```yaml {label="stringConvert"}
+patches:
+  - type: FromCompositeFieldPath
+    fromFieldPath: spec.field1
+    toFieldPath: metadata.annotations["FIELD1-TEXT"]
+    transforms:
+      - type: string
+        string:
+          type: Convert
+          convert: "ToUpper"
+```
+
+#### String format
+The {{<hover label="typeFormat" line="9">}}type: format{{</hover>}}
+applies [Go string formatting](https://pkg.go.dev/fmt) to the input. 
+
+```yaml {label="typeFormat"}
+patches:
+  - type: FromCompositeFieldPath
+    fromFieldPath: spec.field1
+    toFieldPath: metadata.annotations["stringAnnotation"]
+    transforms:
+      - type: string
+        string:
+          type: Format
+          format:
+            fmt: "the-field-%s"
+```
+
+#### Regular expression type
+The {{<hover label="typeRegex" line="8">}}type: Regexp{{</hover>}} extracts
+the part of the input matching a regular expression. 
+
+Optionally use a 
+{{<hover label="typeRegex" line="11">}}group{{</hover>}} to match a regular
+expression capture group.  
+By default Crossplane matches the entire regular expression.
+
+```yaml {label="typeRegex"}
+patches:
+  - type: FromCompositeFieldPath
+    fromFieldPath: spec.desiredRegion
+    toFieldPath: metadata.annotations["euRegion"]
+    transforms:
+      - type: string
+        string:
+          type: Regexp
+          regexp:
+            match: '^eu-(.*)-'
+            group: 1
+```
+
+#### Trim prefix
+
+The {{<hover label="typeRegex" line="8">}}type: TrimPrefix{{</hover>}} removes
+the matching string and all preceding characters. 
+
+```yaml {label="typeRegex"}
+patches:
+  - type: FromCompositeFieldPath
+    fromFieldPath: spec.desiredRegion
+    toFieldPath: metadata.annotations["north-1"]
+    transforms:
+      - type: string
+        string:
+          type: TrimPrefix
+          trim: `eu-
+```
+
+#### Trim suffix
+
+The {{<hover label="typeRegex" line="8">}}type: TrimSuffix{{</hover>}} removes
+the matching string and all proceeding characters. 
+
+```yaml {label="typeRegex"}
+patches:
+  - type: FromCompositeFieldPath
+    fromFieldPath: spec.desiredRegion
+    toFieldPath: metadata.annotations["eu"]
+    transforms:
+      - type: string
+        string:
+          type: TrimSuffix
+          trim: `-north-1'
+```
+
+## Patch policies
+
+Crossplane supports two types of patch policies:
+* `fromFieldPath`
+* `mergeOptions`
+
+<!-- vale Google.Headings = NO -->
+### fromFieldPath policy
+<!-- vale Google.Headings = YES -->
+
+Using a `fromFieldPath: Required` policy on a patch requires the
+`fromFieldPath` to exist in the composite resource.
+
+{{<hint "tip" >}}
+If a resource patch isn't working applying the `fromFieldPath: Required` policy
```

Comment:
should we be explaining `fromFieldPath: Optional` here also? (that's a thing right? 😂)